src/ts/utils/settings.ts

```diff
@@ -41,9 +43,15 @@ export const Settings = {
         (await getStateFromStorage())[featureId][settingId] || defaultValue,
     isActive: async (featureId: string) => (await getStateFromStorage())[featureId]?.active,
 }
+const initDefaultState = (feature: Feature): {active: boolean} => {
+    const active = feature.enabledByDefault === false ? false : true
+    return {
+        active,
+    }
+}
 
 export const prepareSettings = (features: Feature[]): Feature[] => {
-    return features.map((feature: any) => {
+    return features.map((feature: Feature) => {
         if (feature.toggleable !== false) {
             feature.toggleable = true
         }
```

Comment:
in the prev comment about this - I meant more like extracting this, alongside with the similar code you had for `enabledByDefault` to set up the default values on the feature object

Reply:
this is different from that -- this causes side effects by mutating the feature, that does not. let's not mix it up. @Stvad

Reply:
it was before :ton

Reply:
yes :) fixed it